kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -427,6 +433,8 @@ private Node createFieldEntry(PatternField patternField, int fieldNum) {
     }
 
     private void showContextMenuPopUp(PatternField patternField, Node node, double x, double y) {
+        selectedPatternField = patternField;
+        selectedNodeField = node;
```

Comment:
**Question:** why are we creating 2 global variables of a selected field?

My previous comment still applies. Based on the method name, it (`showContextMenuPopUp()`) is doing more than showing the context menu options. Lambdas (on click event listener) does allow you to assign global variables without being `final`. 

Having said this (assuming you have global variables) you have two choices:

1. Remove method and put code into the label's onclick handler
2. Change the name of the function to be `selectPatternField(PatternField patternField);` the onclick handler would call this method and another statement to call `contextMenu.show(label, x, y);`

Reply:
Thanks Carl, I have put the code in setOnMouseClicked and assigned the global variables.